### PR TITLE
Stage Type.equal

### PR DESCRIPTION
**Pull request by samoht**
/cc @CraigFe

### CODE REVIEW

File: src/irmin-pack/IO.ml [hunk 1]
```diff
@@ -106,7 +106,7 @@ module Unix : S = struct
     mutable flushed : int64;
     readonly : bool;
     version : version;
-    buf : Buffer.t;
+    mutable buf : string list;
```
Comment (CraigFe) [inline src/irmin-pack/IO.ml:109]: Looks like an unrelated change here. I got the impression that this ended up being bad for performance?

Comment (samoht) [reply src/irmin-pack/IO.ml:109]: hum yes indeed I though I had reverted this but that doesn't seem to be the case. Thanks for noticing!

**Comment by samoht** [comment]
I have removed the unrelated changes and added a CHANGELOG. Will probably consolidate this file later on anyway, as the generics have a new signature (which is a breaking change).
